tests/test_optimizer.py

```diff
@@ -151,7 +151,9 @@ def test_state_averager(offload_optimizer: bool, reuse_tensors: bool, sync_epoch
         F.mse_loss(model2(x), -torch.ones(3)).backward()
         avgr2.step(optimizer_step=True, zero_grad=True, averaging_round=(step == 10), delay_averaging=False)
 
-    assert torch.all(model1.weight.grad == 0) and torch.all(model2.weight.grad == 0), "zero grad did not trigger"
+    assert (model1.weight.grad is None or torch.all(model1.weight.grad == 0)) and (
+        model2.weight.grad is None or torch.all(model2.weight.grad == 0)
+    ), "zero grad did not trigger"
```

Comment:
Can we also use `Version(torch.__version__).major >= 2` here? (or at least import _SET_TO_NONE_DEFAULT) 

This way, the test won't have a bug where we always set the grad to None regardless of the PyTorch version